crates/core_arch/src/aarch64/tme.rs

```diff
@@ -0,0 +1,124 @@
+pub const _TMFAILURE_REASON: u32 = 0x00007FFF_u32;
+
+/// Transaction retry is possible.
+pub const _TMFAILURE_RTRY: u32 = 1 << 15;
+
+/// Transaction cancelled.
+pub const _TMFAILURE_CNCL: u32 = 1 << 16;
+
+/// Transaction cancelled due to high memory usage.
+pub const _TMFAILURE_MEM: u32 = 1 << 17;
+
+///
+pub const _TMFAILURE_IMP: u32 = 1 << 18;
+
+///
+pub const _TMFAILURE_ERR: u32 = 1 << 19;
+
+///
+pub const _TMFAILURE_SIZE: u32 = 1 << 20;
+
+/// Transaction abort in a inner nested transaction.
+pub const _TMFAILURE_NEST: u32 = 1 << 21;
+
+/// Transaction abort due to a debug trap.
+pub const _TMFAILURE_DBG: u32 = 1 << 22;
+
+///
+pub const _TMFAILURE_INT: u32 = 1 << 23;
+
+///
+pub const _TMFAILURE_TRIVIAL: u32 = 1 << 24;
+
+
+/// Starts a new transaction. When the transaction starts successfully the return value is 0.
+/// If the transaction fails, all state modifications are discarded and a cause of the failure
+/// is encoded in the return value.
+///
+/// [ARM TME Intrinsics](https://developer.arm.com/docs/101028/0010/transactional-memory-extension-tme-intrinsics).
+#[inline]
+#[target_feature(enable = "tme")]
+#[cfg_attr(test, assert_instr(tstart))]
+pub unsafe fn __tstart() -> u32 {
+    aarch64_tstart() as _
+}
+
+/// Commits the current transaction. For a nested transaction, the only effect is that the
+/// transactional nesting depth is decreased. For an outer transaction, the state modifications
+/// performed transactionally are committed to the architectural state.
+///
+/// [ARM TME Intrinsics](https://developer.arm.com/docs/101028/0010/transactional-memory-extension-tme-intrinsics).
+#[inline]
+#[target_feature(enable = "tme")]
+#[cfg_attr(test, assert_instr(tcommit))]
+pub unsafe fn __tcommit() {
+    aarch64_tcommit()
+}
+
+/// Cancels the current transaction and discards all state modifications that were performed transactionally.
+///
+/// [ARM TME Intrinsics](https://developer.arm.com/docs/101028/0010/transactional-memory-extension-tme-intrinsics).
+#[inline]
+#[target_feature(enable = "tme")]
+#[cfg_attr(test, assert_instr(tcancel, imm0 = 0x0))]
+#[rustc_args_required_const(0)]
+pub unsafe fn __tcancel(imm0: u32) {
+    macro_rules! call {
+        ($imm0:expr) => {
+            aarch64_tcancel($imm0)
+        };
+    }
+    constify_imm8!(imm0, call)
+}
+
+/// Tests if executing inside a transaction. If no transaction is currently executing,
+/// the return value is 0. Otherwise, this intrinsic returns the depth of the transaction.
+///
+/// [ARM TME Intrinsics](https://developer.arm.com/docs/101028/0010/transactional-memory-extension-tme-intrinsics).
+#[inline]
+#[target_feature(enable = "tme")]
+#[cfg_attr(test, assert_instr(ttest))]
+pub unsafe fn __ttest() -> u32 {
+    aarch64_ttest() as _
+}
+
+/// Encodes cancellation reason, which is the parameter passed to [`__tcancel`]
+/// Takes cancellation reason flags and retry-ability.
+#[inline]
+pub const fn _tcancel_code(reason: u32, retryable: bool) -> u32 {
```

Comment:
I don't see this in any ARM documentation. As a general rule we try to avoid defining our own helper functions.